microprot/scripts/processing.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+__author__ = "Tomasz Kosciolek"
+__version__ = "1.01b"
+__last_update__ = "12/19/2016"
+
+
+import click
+import sys
+from skbio import io, Sequence
+
+
+def extract_sequences(infile, seqidx=0):
+    """ Extract sequence(s) from a multi-sequence FASTA file
+
+    Parameters
+    ----------
+    infile : str
+        file path to input multi-sequence FASTA file
+    seqidx : int (optional)
+        n-th sequence of the input file to extract
+        (default: 0 for all sequences)
+
+    Returns
+    -------
+    list of skbio Sequence
+    """
+    seqs = []
+    iseq = 0  # current sequence index
+    for seq in io.read(infile, format='fasta'):
+        iseq += 1
+        if 0 < seqidx != iseq:
+            continue
```

Comment:
@1: I guess the reason is lazy evaluation, i.e. skbio returns a **generator** not a **list**. That is a cool feature for very long files. Thus, you can start processing the first sequence before you parsed all say 10 million sequences in that file.

@2: we are prototyping microprot at the moment. Thus, we should focus only on things we need NOW, instead of write cool code that is extendable and general as hell. What is the use case for reading the x-th sequence from a file? Sound like something that a grid scheduler might need to do if you want to run the program on a bunch of sequences.